pkg/auth/client/impersonate.go

```diff
@@ -51,24 +48,3 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
-
-// NewImpersonatingConfig wraps the config's transport to impersonate a user, including user, groups, and scopes
-func NewImpersonatingConfig(user user.Info, config restclient.Config) restclient.Config {
```

Comment:
Unless there is some subtle bug in this func I cannot really see how this would cause issues.